src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/Manager/PluginsManager.cs

```diff
@@ -582,21 +589,37 @@ private async Task<IPluginFetcher> TryGetPluginFetcher(AvailablePluginInfo avail
                     IPluginFetcher fetcher = await TryGetPluginFetcher(pluginInfo);
                     if (fetcher == null)
                     {
-                        PluginSourceErrorOccured?.Invoke(this, new PluginSourceResourceNotFoundEventArgs(source, typeof(IPluginFetcher)));
-                        this.logger.Error($"No fetcher is found that supports fetching plugin {pluginInfo.Identity}. Skipping.");
-                    }
-                    else
-                    {
-                        var newPlugin = new AvailablePlugin(pluginInfo, discoverer, fetcher);
-                        results[pluginInfo.Identity] = (newPlugin, discoverer);
+                        HandleResourceNotFoundError(
+                            source,
+                            $"No fetcher is found that supports fetching plugin {pluginInfo.Identity}.");
+                        continue;
                     }
+                    var newPlugin = new AvailablePlugin(pluginInfo, discoverer, fetcher);
+                    results[pluginInfo.Identity] = (newPlugin, discoverer);
                 }
                 else
                 {
-                    this.logger.Warn($"Duplicate plugin {pluginInfo.Identity} is discovered from {source.Uri} by {discoverer.GetType().Name}." +
+                    this.logger.Warn($"Duplicate plugin {pluginInfo.Identity} is discovered from {source} by {discoverer.GetType().Name}." +
                        $"Using the first found discoverer: {tuple.Item2.GetType().Name}.");
                 } 
             }
         }
+
+
+        private void HandleResourceNotFoundError(PluginSource source, string errorMsg)
+        {
+            var errorInfo = new ErrorInfo(ErrorCodes.PLUGINS_MANAGER_PluginManagerResourceNotFound, errorMsg);
+            PluginSourceErrorOccured?.Invoke(this, new PluginSourceErrorEventArgs(source, errorInfo));
+
+            this.logger.Error(errorMsg);
+        }
+
+        private void HandlePluginSourceException(PluginSource source, string errorMsg, Exception exception)
```

Comment:
I haven't looked at the whole PR yet so I might lack context. I see we added some Exception Error codes too. 

Do you think it might be useful to create a new `PluginException` which extends the Exception class similar to `EngineException` so we can reuse this exception in the future?